File: app/src/main/java/id/rofiqoff/examplefilterdataactivityfragment/java/MainActivity.java
package id.rofiqoff.examplefilterdataactivityfragment.java;

import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.AppCompatEditText;
import androidx.viewpager.widget.ViewPager;

import id.rofiqoff.examplefilterdataactivityfragment.PagerAdapter;
import id.rofiqoff.examplefilterdataactivityfragment.R;

public class MainActivity extends AppCompatActivity {

    private SearchListener searchListener;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();
    }

    private void initView() {
        ViewPager viewPager = findViewById(R.id.view_pager);
        AppCompatEditText editTextSearch = findViewById(R.id.edit_text_search);

        PagerAdapter pagerAdapter = new PagerAdapter(getSupportFragmentManager());
        pagerAdapter.addFragment(new DataFragment(), "");

        viewPager.setAdapter(pagerAdapter);

        editTextSearch.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
            }

            @Override
            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
                if (getDataSearch() != null) {
                    getDataSearch().onSearch(charSequence.toString());
                }
            }

            @Override
            public void afterTextChanged(Editable editable) {
            }
        });
    }

    private SearchListener getDataSearch() {
        return searchListener;
    }

    public void setDataSearch(SearchListener searchListener) {
        this.searchListener = searchListener;
    }

    interface SearchListener {
        void onSearch(String dataToString);
    }

}

File: app/src/main/java/id/rofiqoff/examplefilterdataactivityfragment/kotlin/MainActivity.kt
package id.rofiqoff.examplefilterdataactivityfragment.kotlin

import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import androidx.appcompat.app.AppCompatActivity
import id.rofiqoff.examplefilterdataactivityfragment.PagerAdapter
import id.rofiqoff.examplefilterdataactivityfragment.R
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private var searchDataListener: SearchDataListener? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initView()
    }

    private fun initView() {
        val pageAdapter = PagerAdapter(supportFragmentManager)
        pageAdapter.addFragment(DataFragment())

        view_pager.adapter = pageAdapter

        edit_text_search?.addTextChangedListener(object : TextWatcher {
            override fun afterTextChanged(p0: Editable?) {}
            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {}

            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
                if (getSearchDataListener != null) {
                    getSearchDataListener?.onSearch(p0.toString())
                }
            }

        })
    }

    private val getSearchDataListener: SearchDataListener?
        get() = searchDataListener

    fun setSearchDataListener(searchDataListener: SearchDataListener?) {
        this.searchDataListener = searchDataListener
    }

    interface SearchDataListener {
        fun onSearch(dataToSearch: String)
    }

}
